Source/SwiftLintFramework/Rules/Style/CollectionAlignmentRule.swift

```diff
@@ -0,0 +1,295 @@
+import SourceKittenFramework
+
+public struct CollectionAlignmentRule: ASTRule, ConfigurationProviderRule, OptInRule {
+    public var configuration = CollectionAlignmentConfiguration()
+
+    public init() {}
+
+    public static var description = RuleDescription(
+        identifier: "collection_alignment",
+        name: "Collection Element Alignment",
+        description: "All elements in a collection literal should be vertically aligned",
+        kind: .style,
+        nonTriggeringExamples: Examples(alignColons: false).nonTriggeringExamples,
+        triggeringExamples: Examples(alignColons: false).triggeringExamples
+    )
+
+    public func validate(file: File, kind: SwiftExpressionKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard kind == .dictionary || kind == .array else { return [] }
+
+        let keyLocations: [Location]
+        if kind == .array {
+            keyLocations = getArrayElementLocations(with: file, dictionary: dictionary)
+        } else {
+            keyLocations = getDictionaryKeyLocations(with: file, dictionary: dictionary)
+        }
+
+        guard keyLocations.count >= 2 else {
+            return []
+        }
+
+        let firstKeyLocation = keyLocations[0]
+        let violationLocations = zip(keyLocations[1...], 1...)
```

Comment:
what do you think about using `enumerated` here?

Reply:
`enumerated` would not work out of the box with the `ArraySlice` being created here because `enumerated` is always zero-based. However, I've updated this to use the `indices` property, which I think makes more sense than using `1...`. Thoughts?